frame/sudo/src/mock.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Test utilities
+
+use super::*;
+
+use frame_support::{
+	impl_outer_origin, impl_outer_dispatch, impl_outer_event, parameter_types,
+	weights::Weight,
+	// traits::{OnInitialize, OnFinalize},
+};
+use sp_core::H256;
+// The testing primitives are very useful for avoiding having to work with signatures
+// or public keys. `u64` is used as the `AccountId` and no `Signature`s are required.
+use sp_runtime::{
+	Perbill, traits::{BlakeTwo256, IdentityLookup}, testing::Header,
+};
+use sp_io;
+
+use crate as sudo;
+
+impl_outer_origin! {
+	pub enum Origin for Test where system = frame_system {}
+}
+impl_outer_event! {
+	pub enum TestEvent for Test {
+		system<T>,
+		sudo<T>,
+	}
+}
+impl_outer_dispatch! {
+	pub enum Call for Test where origin: Origin {
+		sudo::Sudo,
+		priveleged_fn_test_module::Priveleged,
+	}
+}
+
+// Dummy module with a privelleged dispatchable function for testing sudo
+mod priveleged_fn_test_module {
+	use frame_support::{decl_module, dispatch};
+	use frame_system::ensure_root;
+	pub trait Trait: frame_system::Trait {}
+
+	decl_module! {
+		pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+			#[weight = 0]
+			pub fn privileged_function(origin) -> dispatch::DispatchResult {
+				ensure_root(origin)?;
+				println!("This is priveleged_fn_test_module::privileged_function() doing something.");
```

Comment:
you should have this function actually do something, like log a value:
https://github.com/paritytech/substrate/blob/master/frame/scheduler/src/lib.rs#L448

Then you can check the state changed when the call is successful.

Furthermore, you shouldn't really use `println` unless you are locally debugging. To commit to substrate, most should be removed.

Reply:
See other comments - to summarize though: dummy module was removed and subbed in with logger module.